source/CIS/cis-4.rst

```diff
@@ -0,0 +1,705 @@
+- The revocation MUST fail if:
+    - The the transaction ``sender`` is not the issuer.
+    - The credential ID is not present in the registry.
+    - The credential status is not one of ``Active`` or ``NotActivated`` (see :ref:`CIS-4-functions-credentialStatus`).
+
+.. _CIS-4-functions-revokeCredentialHolder:
+
+``revokeCredentialHolder``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Revoke a credential by the holders's request.
+
+The holder is authorized to revoke a credential by verifying the signature with the holder's public key.
+It replaces the authorization checks conducted on the ``sender/invoker`` variable with signature verification.
+The public key is part of :ref:`CIS-4-CredentialInfo` that is used when registering a credential with the :ref:`CIS-4-functions-registerCredential` entrypoint.
+
+Parameter
+~~~~~~~~~
+
+It is serialized as :ref:`CIS-4-SignatureEd25519` (``signature``) and message data ``RevocationDataHolder`` consisting of :ref:`CIS-4-CredentialID` (``credential_id``), metadata about the signature :ref:`CIS-4-SigningData` (``signing_data``), and an optional revocation reason (``reason``), serialized similarly to :ref:`CIS-4-functions-revokeCredentialIssuer`::
+
+  RevocationDataHolder ::= (credential_id: CredentialID) (signing_data: SigningData) (reason: OptionReason)
+  RevokeCredentialHolderParam ::= (signature: SignatureEd25519) (data : RevocationDataHolder)
+
+
+Requirements
+~~~~~~~~~~~~
+
+- If revoked successfully, the credential status MUST change to ``Revoked`` (see :ref:`CIS-4-functions-credentialStatus`).
+- The ``RevokeCredentialHolderParam``'s ``signing_data`` MUST include a nonce to protect against replay attacks.
+  The holders's nonce is sequentially increased every time a revocation request is successfully executed.
+  The function MUST only accept a ``RevokeCredentialHolderParam`` if it has the next nonce following the sequential order.
+- The revocation MUST fail if:
+    - The credential ID is not present in the registry.
+    - The credential status is not one of ``Active`` or ``NotActivated`` (see :ref:`CIS-4-functions-credentialStatus`).
+    - The credential is not holder-revocable.
+    - The signature was intended for a different contract.
+    - The signature was intended for a different entrypoint.
+    - The signature is expired.
+    - The signature can not be validated.
+      The smart contract logic SHOULD practice its best efforts to ensure that only the holder can generate and authorize a revocation request with a valid signature.
+
+.. _CIS-4-functions-revokeCredentialOther:
+
+``revokeCredentialOther``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Revoke a credential by a revocation authority request.
+A revocation authority is any entity that holds a private key corresponding to the public key registered by the issuer.
+A revocation authority is authorized to revoke a credential by verifying the signature with the public key this the given identifier.
```

Comment:
Something is off towards the end of this sentence.

Reply:
Fixed.